ethcore/src/blockchain/blockchain.rs

```diff
@@ -466,28 +466,25 @@ impl BlockChain {
 		batch.put(b"best", &update.info.hash).unwrap();
 
 		{
-			let mut write_details = self.block_details.write().unwrap();
-			for (hash, details) in update.block_details.into_iter() {
-				batch.write(&hash, &details);
-				self.note_used(CacheID::Extras(ExtrasIndex::BlockDetails, hash.clone()));
-				write_details.insert(hash, details);
+			for hash in update.block_details.keys().cloned() {
+				self.note_used(CacheID::Extras(ExtrasIndex::BlockDetails, hash));
 			}
+
+			let mut write_details = self.block_details.write().unwrap();
+			BatchWriter::new(&batch, &mut write_details)
```

Comment:
An object is created just to call a single method. Could it be a simple function instead?